### PR TITLE
Fixing warnings

### PR DESCRIPTION
**Pull request by matozoid**
Fix some warnings. Hopefully the changes are not upsetting.

### CODE REVIEW

File: src/main/kotlin/com/strumenta/kolasu/generation/XmlGenerator.kt [hunk 1]
```diff
@@ -123,14 +123,14 @@ private fun Node.toXML(role: String, document: Document): Element {
             if (it.provideNodes) {
                 element.addChild((it.value as Node).toXML(it.name, document))
             } else {
-                element.addAttribute(it.name, it.value, document)
+                element.addAttribute(it.name, it.value)
```
Comment (ftomassetti) [inline src/main/kotlin/com/strumenta/kolasu/generation/XmlGenerator.kt]: +1

File: src/main/kotlin/com/strumenta/kolasu/generation/XmlGenerator.kt [hunk 2]
```diff
@@ -123,14 +123,14 @@ private fun Node.toXML(role: String, document: Document): Element {
...
             }
         }
     }
     return element
 }
 
-private fun Element.addAttribute(role: String, value: Any, document: Document) {
+private fun Element.addAttribute(role: String, value: Any) {
```
Comment (ftomassetti) [inline src/main/kotlin/com/strumenta/kolasu/generation/XmlGenerator.kt]: +1

File: src/main/kotlin/com/strumenta/kolasu/model/Navigation.kt [hunk 3]
```diff
@@ -3,6 +3,7 @@ package com.strumenta.kolasu.model
 /**
  * @return the nearest ancestor of this node that is an instance of klass.
  */
+@Suppress("UNCHECKED_CAST")
```
Comment (ftomassetti) [inline src/main/kotlin/com/strumenta/kolasu/model/Navigation.kt]: Should we add a comment to explain why this is ok here?

File: src/main/kotlin/com/strumenta/kolasu/model/Printing.kt [hunk 4]
```diff
@@ -9,31 +9,32 @@ private const val indentBlock = "  "
 fun Node.relevantMemberProperties() = this.javaClass.kotlin.memberProperties
     .filter { !it.name.startsWith("component") && it.name != "position" && it.name != "parent" }
 
+@Suppress("UNCHECKED_CAST")
```
Comment (ftomassetti) [inline src/main/kotlin/com/strumenta/kolasu/model/Printing.kt]: Should we add a comment to explain why this is ok here?

File: src/main/kotlin/com/strumenta/kolasu/model/Printing.kt [hunk 5]
```diff
@@ -9,31 +9,32 @@ private const val indentBlock = "  "
...
     if (this.relevantMemberProperties().isEmpty()) {
         sb.append("$indent${this.javaClass.simpleName}\n")
     } else {
         sb.append("$indent${this.javaClass.simpleName} {\n")
-        this.relevantMemberProperties().forEach {
-            val mt = it.returnType.javaType
-            if (mt is ParameterizedType && mt.rawType.equals(List::class.java)) {
+        this.relevantMemberProperties().forEach { property ->
+            val mt = property.returnType.javaType
+            if (mt is ParameterizedType && mt.rawType == List::class.java) {
```
Comment (ftomassetti) [inline src/main/kotlin/com/strumenta/kolasu/model/Printing.kt]: right!

File: src/main/kotlin/com/strumenta/kolasu/model/Processing.kt [hunk 6]
```diff
@@ -145,6 +145,7 @@ fun Node.find(predicate: (Node) -> Boolean): Node? {
 /**
  * Recursively execute "operation" on this node, and all nodes below this node that extend klass.
  */
+@Suppress("UNCHECKED_CAST")
```
Comment (ftomassetti) [inline src/main/kotlin/com/strumenta/kolasu/model/Processing.kt]: Should we add a comment to explain why this is ok here?

**Comment by ftomassetti** [comment]
...but if you say "no", I will merge as it is

File: src/main/kotlin/com/strumenta/kolasu/model/Navigation.kt (hunk 3 above)
Comment (matozoid) [reply src/main/kotlin/com/strumenta/kolasu/model/Navigation.kt]: This one is gone when we merge the node_sequences branch.

File: src/main/kotlin/com/strumenta/kolasu/model/Processing.kt (hunk 6 above)
Comment (matozoid) [reply src/main/kotlin/com/strumenta/kolasu/model/Processing.kt]: This one probably also disappears.

File: src/main/kotlin/com/strumenta/kolasu/model/Printing.kt (hunk 5 above)
Comment (matozoid) [reply src/main/kotlin/com/strumenta/kolasu/model/Printing.kt]: Yeah, I couldn't read the code so I explicitly called things "property" and "value" after a while. What's an "mt"? ;-)

The equals was just sitting there in the code, all yellow.

**Pull request merged**